idf_build_apps/manifest/manifest.py

```diff
@@ -51,6 +48,9 @@ def __init__(self, stmt: str, temporary: bool = False, reason: t.Optional[str] =
                 f'    reason: lack of ci runners'
             )
 
+    def __repr__(self):
+        return f'IfClause(stmt={self._stmt!r}, temporary={self.temporary!r}, reason={self.reason!r})'
```

Comment:
```suggestion
        return f'IfClause(stmt={self._stmt}, temporary={self.temporary}, reason={self.reason})'
```
These're base data structures, no need to format with `repr`

Reply:
For `_stmt`, I agree, but for the optional string (reason), not really, because it will return different results.
```
>>> repr(None)
'None'
>>> str(None)
'None'
>>> str('None')
'None'
>>> repr('None')
"'None'"
```

Reply:
I see. make sense!

Reply:
Actually, I also disagree with `self._stmt`:
- With `!r`: `[IfClause(stmt='IDF_TARGET == "esp32c3"', temporary=False, reason='None')]`
- Without `!r`: `[IfClause(stmt=IDF_TARGET == "esp32c3", temporary=False, reason='None')]`"